neuro_san/client/direct_agent_session_factory.py

```diff
@@ -60,46 +59,46 @@ def create_session(self, agent_name: str, use_direct: bool = False,
                          dictionary of string keys to string values.
         """
 
-        tool_registry: AgentToolRegistry = self.get_agent_tool_registry(agent_name)
+        agent_network: AgentNetwork = self.get_agent_network(agent_name)
 
         llm_factory: ContextTypeLlmFactory = MasterLlmFactory.create_llm_factory()
         toolbox_factory: ContextTypeToolboxFactory = MasterToolboxFactory.create_toolbox_factory()
         # Load once now that we know what tool registry to use.
         # Include "agent_llm_info_file" from agent network hocon to llm factory.
-        agent_llm_info_file = tool_registry.get_agent_llm_info_file()
+        agent_llm_info_file = agent_network.get_agent_llm_info_file()
         llm_factory.load(agent_llm_info_file)
         toolbox_factory.load()
 
         factory = ExternalAgentSessionFactory(use_direct=use_direct)
         invocation_context = SessionInvocationContext(factory, llm_factory, toolbox_factory, metadata)
         invocation_context.start()
-        session: DirectAgentSession = DirectAgentSession(tool_registry=tool_registry,
+        session: DirectAgentSession = DirectAgentSession(agent_network=agent_network,
```

Comment:
DirectAgentSession and its async buddy now take an AgentNetwork (config only), not an AgentToolRegistry (config + factory).  This way the session/service level doesn't really have to lug around objects that deal with factory aspects at all. FWIW: The transition to needing a factory comes right when it needs to in DataDrivenChat (see below).